usr/lib/linuxmint/mintUpdate/automatic_upgrades.py

```diff
@@ -6,45 +6,54 @@
 
 optionsfile = "/etc/mintupdate-automatic-upgrades.conf"
 logfile = "/var/log/mintupdate.log"
+power_connectfile="/sys/class/power_supply/AC/online"
 log = open(logfile, "a")
 log.write("\n-- Automatic Upgrade starting %s:\n" % time.strftime('%a %d %b %Y %H:%M:%S %Z'))
 log.flush()
 
 pkla_source = "/usr/share/linuxmint/mintupdate/automation/99-mintupdate-temporary.pkla"
 pkla_target = "/etc/polkit-1/localauthority/90-mandatory.d/99-mintupdate-temporary.pkla"
 try:
-    # Put shutdown and reboot blocker into place
-    os.symlink(pkla_source, pkla_target)
+    powersupply = open(power_connectfile).read()[0]=='1'
```

Comment:
can you close power_connectfile after reading it? Or use a context manager, which will close it automatically.

like:
```
with open(power_connectfile) as f:
    powersupply = f.read()[0]
```